source/kraken/realtime.cpp

```diff
@@ -281,6 +256,76 @@ static bool is_added_stop_time(const transit_realtime::TripUpdate& trip_update)
     return false;
 }
 
+static bool is_handleable(const transit_realtime::TripUpdate& trip_update,
+                          const nt::PT_Data& pt_data,
+                          const bool is_realtime_add_enabled,
+                          const bool is_realtime_add_trip_enabled)
+{
+    namespace bpt = boost::posix_time;
+
+    auto log = log4cplus::Logger::getInstance("realtime");
+
+    // if is_realtime_add_enabled = false, this disables the adding trip option too
+    if ((! is_realtime_add_enabled && is_added_stop_time(trip_update)) ||
+        (! is_realtime_add_enabled && is_added_trip(trip_update))) {
+        LOG4CPLUS_DEBUG(log, "Add stop time is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDED/ADDED_FOR_DETOUR stop times or ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    if (! is_realtime_add_trip_enabled && is_added_trip(trip_update)) {
+        LOG4CPLUS_DEBUG(log, "Add trip is disabled: ignoring trip update id "
+                << trip_update.trip().trip_id()
+                << " with ADDITIONAL_SERVICE effect");
+        return false;
+    }
+    // Case 1: the trip is cancelled
+    if (trip_update.trip().schedule_relationship() == transit_realtime::TripDescriptor_ScheduleRelationship_CANCELED) {
+        return true;
+    }
+    // Case 2: the trip is not cancelled, but modified (ex: the train's arrival is delayed or the train's
+    // departure is brought forward), we check the size of stop_time_update because we can't find a proper
+    // enum in gtfs-rt proto to express this idea
+    if (is_circulating(trip_update)) {
+
+        // check if company id exists
+        if (trip_update.trip().HasExtension(kirin::company_id) &&
+            pt_data.companies_map.find(trip_update.trip().GetExtension(kirin::company_id)) == pt_data.companies_map.end()) {
+            LOG4CPLUS_DEBUG(log, "Trip company id "
+                    << trip_update.trip().GetExtension(kirin::company_id)
+                    <<  " doesn't exist: ignoring trip update id "
+                    << trip_update.trip().trip_id());
+            return false;
+        }
+        // check if physical mode id exists
+        if (trip_update.vehicle().HasExtension(kirin::physical_mode_id) &&
+            pt_data.physical_modes_map.find(trip_update.vehicle().GetExtension(kirin::physical_mode_id)) == pt_data.physical_modes_map.end()) {
+            LOG4CPLUS_DEBUG(log, "Trip physical mode id "
+                    << trip_update.vehicle().GetExtension(kirin::physical_mode_id)
+                    <<  " doesn't exist: ignoring trip update id "
+                    << trip_update.trip().trip_id());
+            return false;
```

Comment:
Maybe we can fallback on `physical_mode:Coach` if it exists (most common case considering SNCF or other transport providers).
Maybe we can do this fallback in apply_disruption directly.

Reply:
For the moment if we don't match with the key, as we said, we reject. What is the problem with that ?

Reply:
:heavy_check_mark: Discussed and agreed on another feed (company+physical_mode management)